classes/robot/crawler.php

```diff
@@ -338,6 +338,9 @@ public function mark_for_crawl($baseurl, $url, $courseid = null, $priority = TOO
             $excludes = str_replace("\r", '', self::get_config()->excludeexturl);
         }
 
+        $excludes = explode( PHP_EOL, $excludes);
```

Comment:
Is this required?

```
lib/xhprof/xhprof_moodle.php:467:function profiling_string_matches($string, $patterns) {
lib/xhprof/xhprof_moodle.php-468-   $patterns = preg_split("/\n|,/", $patterns);
```

Reply:
This is existing function

Reply:
$excludes = explode( PHP_EOL, $excludes); is required as pattern match happens after for individual url